scripts/west_commands/bindesc.py

```diff
@@ -0,0 +1,315 @@
+
+        list_parser = subparsers.add_parser('list', help='List all known descriptors')
+        list_parser.set_defaults(subcmd='list', big_endian=False)
+
+        return parser
+
+    def dump(self, args):
+        image = self.get_image_data(args.file)
+
+        descriptors = self.parse_descriptors(image)
+        for tag, value in descriptors.items():
+            if tag in self.TAG_TO_NAME:
+                tag = self.TAG_TO_NAME[tag]
+            log.inf(f'{tag}', self.bindesc_repr(value))
+
+    def list(self, args):
+        for tag in self.TAG_TO_NAME.values():
+            log.inf(f'{tag}')
+
+    def common_search(self, args, search_term):
+        image = self.get_image_data(args.file)
+
+        descriptors = self.parse_descriptors(image)
+
+        if search_term in descriptors:
+            value = descriptors[search_term]
+            log.inf(self.bindesc_repr(value))
+        else:
+            log.die('Descriptor not found')
+
+    def search(self, args):
+        try:
+            search_term = self.NAME_TO_TAG[args.descriptor]
+        except KeyError:
+            log.die(f'Descriptor {args.descriptor} is invalid')
+
+        self.common_search(args, search_term)
+
+    def custom_search(self, args):
+        custom_type = {
+            'STR': self.TYPE_STR,
+            'UINT': self.TYPE_UINT,
+            'BYTES': self.TYPE_BYTES
+        }[args.type]
+        custom_tag = self.bindesc_gen_tag(custom_type, int(args.id, 16))
+        self.common_search(args, custom_tag)
+
+    def do_run(self, args, _):
+        if MISSING_REQUIREMENTS:
+            raise RuntimeError('one or more Python dependencies were missing; '
+                               'see the getting started guide for details on '
+                               'how to fix')
+        self.is_big_endian = args.big_endian
+        self.file_type = self.guess_file_type(args)
+        subcmd = getattr(self, args.subcmd)
+        subcmd(args)
+
+    def get_image_data(self, file_name):
+        if self.file_type == 'bin':
+            with open(file_name, 'rb') as bin_file:
+                return bin_file.read()
+
+        if self.file_type == 'hex':
+            return IntelHex(file_name).tobinstr()
+
+        if self.file_type == 'uf2':
+            with open(file_name, 'rb') as uf2_file:
+                return convert_from_uf2(uf2_file.read())
+
+        if self.file_type == 'elf':
+            with open(file_name, 'rb') as f:
+                elffile = ELFFile(f)
+
+                section = elffile.get_section_by_name('rom_start')
+                if section:
+                    return section.data()
+
+                section = elffile.get_section_by_name('text')
+                if section:
+                    return section.data()
+
+            log.die('No "rom_start" or "text" section found')
+
+        log.die('Unknown file type')
+
+    def parse_descriptors(self, image):
```

Comment:
This probably needs to be more resilient to arbitrary hex block. The MAGIC sequence can appear in normal code and data outside of the binary descriptor block, especially in a bin file where you search through the whole binary. Or that a file with this MAGIC but not built with binary descriptor, etc... etc... etc. Probably a good idea to create a future enhancement ticket in the same time so we will remember to amend this.

Reply:
Right, though the odds of a collision are pretty low as the descriptors are always very close to the beginning of the file (unless it's a native posix elf or a hex with multiple images).
I don't think this is much of a problem, because the odds are low and the script is not used for production. For the "real" use of this feature, which is reading the descriptors between partitions (will hopefully be added in a later PR), the offset will be known, and there will be no need to search for the magic.
If you still find it problematic, I can think of two solutions. The first is to add an offset argument to the script, which will make it much harder to use. The second is to make the magic much longer to make the odds of a collision negligible. In that case the change should be made now, to not break compatibility.

Reply:
A 64-bit magic can encode `BINDESC`. Makes it easier to look at the binary using hex editor. :)

Though, it is up to you. I am good either way.

Reply:
I made the magic 64 bits long, though I kept it a random number (`BINDESC.` may turn up in a native posix ELF, in the file name or debug symbols or whatever)